patches/0011-implement-ms_tls_config_schannel-experiment.patch

```diff
@@ -0,0 +1,357 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: qmuntal <qmuntaldiaz@microsoft.com>
+Date: Thu, 3 Jul 2025 15:13:26 +0200
+Subject: [PATCH] implement ms_tls_config_schannel experiment
+
+---
+ src/crypto/tls/cipher_suites_windows.go       | 74 +++++++++++++++++++
+ src/crypto/tls/cipher_suites_windows_test.go  | 67 +++++++++++++++++
+ src/crypto/tls/fips140_test.go                |  7 ++
+ src/crypto/tls/tls_test.go                    |  8 ++
+ .../exp_ms_tls_config_schannel_off.go         |  8 ++
+ .../exp_ms_tls_config_schannel_on.go          |  8 ++
+ src/internal/goexperiment/flags.go            |  4 +
+ .../syscall/windows/security_windows.go       | 28 +++++++
+ .../syscall/windows/zsyscall_windows.go       | 16 ++++
+ 9 files changed, 220 insertions(+)
+ create mode 100644 src/crypto/tls/cipher_suites_windows.go
+ create mode 100644 src/crypto/tls/cipher_suites_windows_test.go
+ create mode 100644 src/internal/goexperiment/exp_ms_tls_config_schannel_off.go
+ create mode 100644 src/internal/goexperiment/exp_ms_tls_config_schannel_on.go
+
+diff --git a/src/crypto/tls/cipher_suites_windows.go b/src/crypto/tls/cipher_suites_windows.go
+new file mode 100644
+index 00000000000000..c1dcd97555fe4d
+--- /dev/null
++++ b/src/crypto/tls/cipher_suites_windows.go
+@@ -0,0 +1,74 @@
++// Copyright 2025 The Go Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style
++// license that can be found in the LICENSE file.
++
++//go:build goexperiment.ms_tls_config_schannel
++
++package tls
++
++import (
++	"internal/syscall/windows"
++	"slices"
++	"unsafe"
++)
++
++func init() {
++	cipherSuitesSchannel, err := cipherSuitesSchannel()
++	if err != nil {
++		panic(err) // This should never happen, as Schannel is always available on Windows.
++	}
++	cipherSuitesPreferenceOrder = filterCipherSuites(cipherSuitesSchannel, cipherSuitesPreferenceOrder)
++	cipherSuitesPreferenceOrderNoAES = filterCipherSuites(cipherSuitesSchannel, cipherSuitesPreferenceOrderNoAES)
++	defaultCipherSuitesTLS13 = filterCipherSuites(cipherSuitesSchannel, defaultCipherSuitesTLS13)
++	defaultCipherSuitesTLS13NoAES = filterCipherSuites(cipherSuitesSchannel, defaultCipherSuitesTLS13NoAES)
++	allowedCipherSuitesFIPS = filterCipherSuites(cipherSuitesSchannel, allowedCipherSuitesFIPS)
++	allowedCipherSuitesTLS13FIPS = filterCipherSuites(cipherSuitesSchannel, allowedCipherSuitesTLS13FIPS)
++}
++
++func cipherSuiteID(name string) (uint16, bool) {
++	for _, c := range CipherSuites() {
++		if c.Name == name {
++			return c.ID, true
++		}
++	}
++	for _, c := range InsecureCipherSuites() {
++		if c.Name == name {
++			return c.ID, true
++		}
++	}
++	return 0, false
++}
++
++// cipherSuitesSchannel returns all the cipher suites that Schannel supports in preference order.
```

Comment:
```suggestion
+// cipherSuitesSchannel returns all the cipher suites that Schannel supports in Schannel's preference order.
```

(The comment on the expeirment makes it clear that it should be this way, so just bringing that info a little closer to the implementation.)